bauhaus/core.py

```diff
@@ -62,15 +63,31 @@ def __repr__(self) -> str:
     def purge_propositions(self):
         """ Purges the propositional variables of an Encoding object """
         self.propositions = defaultdict(weakref.WeakValueDictionary)
-    
+
     def clear_constraints(self):
         """ Clears the constraints of an Encoding object """
         self.constraints = set()
-    
+
     def clear_debug_constraints(self):
         """Clear debug_constraints attribute in Encoding"""
         self.debug_constraints = dict()
 
+    def add_constraint(self, cons: NNF):
+        """Add an NNF constraint to the encoding.
+
+        Arguments
+        ---------
+        cons : NNF
+            Constraint to be added.
+        """
+        assert self._raw_constraints is not None, \
```

Comment:
should we rename raw_constraints as custom_constraints? i think it could be more clear to a user what its purpose is and it also matches the naming of the CustomNNF class